### PR TITLE
Is obstructed

### PR DESCRIPTION
**Pull request by chris-campbell**
I put together a logic for the is_obstructed? method. It's still incomplete because it requires a board model to complete the rspec testing. It's basically a base to work from regarding piece model.

### CODE REVIEW

File: app/models/piece.rb [hunk 1]
```diff
@@ -1,2 +1,80 @@
 class Piece < ApplicationRecord
+  belongs_to :game
+
+  def occupied?(x, y)
+    board.pieces.where(x_position: x, y_position: y).present?
```
Comment (ashsidhu) [inline app/models/piece.rb]: Fix schema, then fix this.

File: app/models/piece.rb [hunk 2]
```diff
@@ -1,2 +1,80 @@
...
+  end
+
+  def obstructed?(end_x, end_y)
+    horizontal_obstruct?(end_x, end_y)
+    vertical_obstruct?(end_x, end_y)
```
Comment (ashsidhu) [inline app/models/piece.rb]: vertical_obstruct?(end_y)
horizontal_obstruct?(end_x)

File: app/models/piece.rb [hunk 3]
```diff
@@ -1,2 +1,80 @@
...
+  end
+  
+  def vertical_obstruct?(_end_x, end_y)
+    vertical_down_obstruct?(end_x, end_y)
+    vertical_up_obstruct?(end_x, end_y)
+  end
+
+  # Checks btw piece and desired position for obstruction on vertical.
+  def vertical_down_obstruct?(_end_x, end_y)
+    if position_y < end_y # checks vertical down
```
Comment (ashsidhu) [inline app/models/piece.rb]: @position_x

File: app/models/piece.rb (hunk 2 above)
Comment (chris-campbell) [reply app/models/piece.rb]: task completed

**Pull request merged**